test/unit/decode_fixtures.test.cpp

```diff
@@ -186,9 +186,169 @@ TEST_CASE("Read feature multipolygon")
+    CHECK(fm.empty());
+}
+
+TEST_CASE("Read: missing geometry type field, results in feature collection and no layer")
 {
     std::string buffer = open_tile("test/mvt-fixtures/fixtures/003/tile.mvt");
     auto fm = mapbox::vector_tile::decode_tile<std::int64_t>(buffer);
     CHECK(fm.empty());
 }
+
+TEST_CASE("Read: missing layer extent is assigned the default value")
+{
+    std::string buffer = open_tile("test/mvt-fixtures/fixtures/009/tile.mvt");
+    auto fm = mapbox::vector_tile::decode_tile<std::int64_t>(buffer);
+    REQUIRE(fm.size() == 1);
+    REQUIRE(fm.end() != fm.find("hello"));
+    auto fc = fm["hello"];
+    REQUIRE(fc.size() == 1);
+    auto f = fc[0];
+    REQUIRE(f.geometry.is<mapbox::geometry::point<std::int64_t>>());
+    auto pt = f.geometry.get<mapbox::geometry::point<std::int64_t>>();
+    // just creating a point is a check that we are handling with a default extent,
+    // but we'll want to assert on extent somewhere in the future
+}
+
+TEST_CASE("Read: A Layer value property is listed as 'string' but encoded as std::int64_t, results in a skip and no properties on feature")
+{
+    std::string buffer = open_tile("test/mvt-fixtures/fixtures/010/tile.mvt");
+    auto fm = mapbox::vector_tile::decode_tile<std::int64_t>(buffer);
+    REQUIRE(fm.size() == 1);
+    REQUIRE(fm.end() != fm.find("hello"));
+    auto fc = fm["hello"];
+    REQUIRE(fc.size() == 1);
+    auto f = fc[0];
+    CHECK(f.properties.empty());
+}
+
+TEST_CASE("Read: two layers with the same name value, but only the first layer added is kept")
+{
+    std::string buffer = open_tile("test/mvt-fixtures/fixtures/015/tile.mvt");
+    auto fm = mapbox::vector_tile::decode_tile<std::int64_t>(buffer);
+    REQUIRE(fm.size() == 1);
+    // right now we default to the std::map::emplace functionality, which will automatically
+    // remove an element where the key currently exists - meaning the first layer added
+    // will be the final layer. Is this what we want? Can we override the feature
+    // collection map to throw an exception? Something to benchmark and think about.
+    REQUIRE(fm.size() == 1);
+    REQUIRE(fm.end() != fm.find("hello"));
+    auto fc = fm["hello"];
+    REQUIRE(fc.size() == 1);
+    auto f = fc[0];
+    REQUIRE(f.geometry.is<mapbox::geometry::point<std::int64_t>>());
+    auto pt = f.geometry.get<mapbox::geometry::point<std::int64_t>>();
+    CHECK(pt.x == 25);
+    CHECK(pt.y == 17);
+    // it would be better to assert on a property that is unique, rather than a geometry that is unique
+}
+
+TEST_CASE("Read: missing layer version property, defaults to a specific version and properly parses")
+{
+    std::string buffer = open_tile("test/mvt-fixtures/fixtures/024/tile.mvt");
+    auto fm = mapbox::vector_tile::decode_tile<std::int64_t>(buffer);
+    CHECK(!fm.empty());
+    // not exactly sure how to test that this one works other than making sure decode_tile actually works
```

Comment:
Any ideas here @flippmoke?

Reply:
This is the correct way to handle this test. You could get into my detail to check specific information on the test if needed, but I am not sure it is.